src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,413 @@
+
+                case ' ': case '\t': case '/': case '\r': case '\n':
+                    // trivia cases
+                    return true;
+            }
+
+            // more trivia
+            if (char.IsWhiteSpace(ch))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private (ImmutableArray<VirtualChar>, JsonKind, EmbeddedDiagnostic? diagnostic) ScanNextTokenWorker()
+        {
+            Debug.Assert(Position < Text.Length);
+            switch (this.CurrentChar)
+            {
+                case '{': return ScanSingleCharToken(JsonKind.OpenBraceToken);
+                case '}': return ScanSingleCharToken(JsonKind.CloseBraceToken);
+                case '[': return ScanSingleCharToken(JsonKind.OpenBracketToken);
+                case ']': return ScanSingleCharToken(JsonKind.CloseBracketToken);
+                case '(': return ScanSingleCharToken(JsonKind.OpenParenToken);
+                case ')': return ScanSingleCharToken(JsonKind.CloseParenToken);
+                case ',': return ScanSingleCharToken(JsonKind.CommaToken);
+                case ':': return ScanSingleCharToken(JsonKind.ColonToken);
+
+                case '\'': case '"':
+                    return ScanString();
+
+                //case '-': case '.':
+                //case '0': case '1': case '2': case '3': case '4':
+                //case '5': case '6': case '7': case '8': case '9':
+                //    return ScanNumber();
+
+                default:
+                    return ScanText();
+            }
+        }
+
+        private (ImmutableArray<VirtualChar>, JsonKind, EmbeddedDiagnostic?) ScanString()
+        {
+            var start = Position;
+            var startChar = this.CurrentChar.Char;
+            Position++;
+
+            EmbeddedDiagnostic? diagnostic = null;
+            while (Position < Text.Length)
+            {
+                var currentCh = this.CurrentChar.Char;
+
+                Position++;
+                switch (currentCh)
+                {
+                    case '"':
+                    case '\'':
+                        if (currentCh == startChar)
+                        {
+                            return (GetSubPattern(start, Position), JsonKind.StringToken, diagnostic);
+                        }
+                        continue;
+
+                    case '\\':
+                        var escapeDiag = ScanEscape(start, Position - 1);
+                        diagnostic = diagnostic ?? escapeDiag;
+                        continue;
+                }
+            }
+
+            var chars = GetSubPattern(start, Position);
+            diagnostic = diagnostic ?? new EmbeddedDiagnostic(
+                WorkspacesResources.Unterminated_string, GetSpan(chars));
+            return (chars, JsonKind.StringToken, diagnostic);
+        }
+
+        private EmbeddedDiagnostic? ScanEscape(int stringStart, int escapeStart)
```

Comment:
`ScanEscape` could use a comment since unlike other scan methods, it doesn't return tokens or nodes, it just advances the position.

Reply:
done.  #Resolved